### PR TITLE
[#375] ci decoupling

### PR DESCRIPTION
**Pull request by alexander-semenets**

### CODE REVIEW

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test this

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test app

File: deploy/ci-infra.Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        stage('Terminate Legion Enclave') {
+            result = build job: params.TerminateLegionEnclaveJobName, propagate: true, wait: true, parameters: [
+                    [$class: 'GitParameterValue', name: 'GitBranch', value: params.GitBranch],
+                    string(name: 'Profile', value: params.Profile),
+                    string(name: 'EnclaveName', value: 'enclave-ci')
+            ]
+        }
+
+        stage('Terminate Cluster') {
```
Comment (dsuslov) [inline deploy/ci-infra.Jenkinsfile]: @alexander-semenets move terminate cluster to finally section

File: deploy/ci-infra.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
+def legionVersion = null
+
+node {
+    try {
+        stage('Terminate Cluster if exists') {
```
Comment (dsuslov) [inline deploy/ci-infra.Jenkinsfile]: @alexander-semenets remove it since cluster will be terminated in finally section

**Comment by alexander-semenets** [comment]
bot, test infra

File: deploy/ci-infra.Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,64 @@
+def legionVersion = "latest"
+
+node {
+    try {
+        stage('Terminate Cluster if exists') {
```
[the remark this answers is not in the document]
Comment (alexander-semenets) [reply deploy/ci-infra.Jenkinsfile]: No warranty that someone else has not started it for some reason, or the previous job was not aborted or crashed, or smth else happened. This step doesn't take too much time (few mins max) but with this check we make the process more reliable.

File: deploy/ci-full-lifecycle.Jenkinsfile [hunk 4]
```diff
@@ -94,19 +95,19 @@ node {
...
             ]
         }
-
+    }
+    catch (e) {
+        // If there was an exception thrown, the build failed
+        currentBuild.result = "FAILED"
+        throw e
+    }
+    finally {
```
[the remark this answers is not in the document]
Comment (alexander-semenets) [reply deploy/ci-full-lifecycle.Jenkinsfile]: pls see the comment above

File: deploy/ci-infra.Jenkinsfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexander-semenets) [reply deploy/ci-infra.Jenkinsfile]: UPDATE:
agreed to replace force termination with step which will check if cluster by some reason exists and throw an error for further investigation. Should be implemented in a separate ticket.
@dsuslov

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test infra

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Pull request merged**